### ISSUE TITLE
Research menu and menubar accessibility

### ISSUE THREAD

**Issue by nakayan5** [yamada-ui/yamada-ui#1501]
### Description

read [this article](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/) and research menubar accessibility

modify component if necessary

### Problem Statement/Justification

none

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

**Comment by nakayan5** [comment]
@yamada-ui/menu is applicable

Keyboard Interaction
- [ ] The following description of keyboard behaviors assumes:
  - [ ] A horizontal menubar containing several menuitem, menuitemradio, or menuitemcheckbox elements.
  - [ ] Some menuitem elements in the menubar have child submenus that contain vertically arranged items.
  - [ ] Some of the menuitem elements in the submenus have child submenus with items that are also vertically arranged.
- [ ] When reading the following descriptions, also keep in mind that:
  - [ ] Focusable elements, which may have role menuitem, menuitemradio, or menuitemcheckbox, are referred to as items.
  - [ ] If a behavior applies to only certain types of items, e.g., menuitem elements, the specific role name is used.
  - [ ] Submenus, also known as popup menus, are elements with role menu.
  - [ ] Except where noted, menus opened from a menubutton behave the same as menus opened from a menubar.

**Comment by nakayan5** [comment]
WAI-ARIA Roles, States, and Properties
- [ ] A menu is a container of items that represent choices. The element serving as the menu has a role of either [menu](https://w3c.github.io/aria/#menu) or [menubar](https://w3c.github.io/aria/#menubar).
- [ ] The items contained in a menu are child elements of the containing menu or menubar and have any of the following roles:
  - [ ] [menuitem](https://w3c.github.io/aria/#menuitem)
  - [ ] [menuitemcheckbox](https://w3c.github.io/aria/#menuitemcheckbox)
  - [ ] [menuitemradio](https://w3c.github.io/aria/#menuitemradio)
- [ ] If activating a [menuitem](https://w3c.github.io/aria/#menuitem) opens a submenu, the menuitem is known as a parent menuitem. A submenu's menu element is:
  - [ ] Contained inside the same menu element as its parent menuitem.
  - [ ] Is the sibling element immediately following its parent menuitem.
- [ ] A parent menuitem has [aria-haspopup](https://w3c.github.io/aria/#aria-haspopup) set to either menu or true.
- [ ] A parent menuitem has [aria-expanded](https://w3c.github.io/aria/#aria-expanded) set to false when its child menu is not visible and set to true when the child menu is visible.
- [ ] One of the following approaches is used to enable scripts to move focus among items in a menu as described in the practice for [Keyboard Navigation Inside Components](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_general_within):
  - [ ] The menu container has tabindex set to -1 or 0 and [aria-activedescendant](https://w3c.github.io/aria/#aria-activedescendant) set to the ID of the focused item.
  - [ ] Each item in the menu has tabindex set to -1, except in a menubar, where the first item has tabindex set to 0.
- [ ] When a [menuitemcheckbox](https://w3c.github.io/aria/#menuitemcheckbox) or [menuitemradio](https://w3c.github.io/aria/#menuitemradio) is checked, [aria-checked](https://w3c.github.io/aria/#aria-checked) is set to true.
- [ ] When a menu item is disabled, [aria-disabled](https://w3c.github.io/aria/#aria-disabled) is set to true.
- [ ] Items in a menu may be divided into groups by placing an element with a role of [separator](https://w3c.github.io/aria/#separator) between groups. For example, this technique should be used when a menu contains a set of [menuitemradio](https://w3c.github.io/aria/#menuitemradio) items.
- [ ] All [separators](https://w3c.github.io/aria/#separator) should have [aria-orientation](https://w3c.github.io/aria/#aria-orientation) consistent with the separator's orientation.
- [ ] If a menubar has a visible label, the element with role menubar has [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set to a value that refers to the labelling element. Otherwise, the menubar element has a label provided by [aria-label](https://w3c.github.io/aria/#aria-label).
- [ ] If a menubar is vertically oriented, it has [aria-orientation](https://w3c.github.io/aria/#aria-orientation) set to vertical. The default value of aria-orientation for a menubar is horizontal.
- [ ] An element with role menu either has:
  - [ ] [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) set to a value that refers to the menuitem or button that controls its display.
  - [ ] A label provided by [aria-label](https://w3c.github.io/aria/#aria-label).
- [ ] If a menu is horizontally oriented, it has [aria-orientation](https://w3c.github.io/aria/#aria-orientation) set to horizontal. The default value of aria-orientation for a menu is vertical.

**Comment by nakayan5** [comment]
related https://github.com/yamada-ui/yamada-ui/issues/1502

**Comment by nakayan5** [comment]
@koralle 
please double check
